docs/calling-downstream-apis/custom-apis.md

```diff
@@ -0,0 +1,823 @@
+# Calling Custom APIs
+
+This guide explains the different approaches for calling your own protected APIs using Microsoft.Identity.Web: IDownstreamApi, IAuthorizationHeaderProvider, and MicrosoftIdentityMessageHandler.
+
+## Overview
+
+When calling custom REST APIs, you have three main options depending on your needs:
+
+| Approach | Complexity | Flexibility | Use Case |
+|----------|------------|-------------|----------|
+| **IDownstreamApi** | Low | Medium | Standard REST APIs with configuration |
+| **MicrosoftIdentityMessageHandler** | Medium | High | HttpClient with DI and composable pipeline |
+| **IAuthorizationHeaderProvider** | High | Very High | Complete control over HTTP requests |
+
+## IDownstreamApi - Recommended for Most Scenarios
+
+`IDownstreamApi` provides a simple, configuration-driven approach for calling REST APIs with automatic token acquisition.
+
+### Installation
+
+```bash
+dotnet add package Microsoft.Identity.Web.DownstreamApi
+```
+
+### Configuration
+
+Define your API in appsettings.json:
+
+```json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "TenantId": "your-tenant-id",
+    "ClientId": "your-client-id",
+    "ClientCredentials": [
+      {
+        "SourceType": "ClientSecret",
+        "ClientSecret": "your-client-secret"
+      }
+    ]
+  },
+  "DownstreamApis": {
+    "MyApi": {
+      "BaseUrl": "https://api.example.com",
+      "Scopes": ["api://my-api-client-id/read", "api://my-api-client-id/write"],
+      "RelativePath": "api/v1",
+      "RequestAppToken": false
+    },
+    "PartnerApi": {
+      "BaseUrl": "https://partner.example.com",
+      "Scopes": ["api://partner-api-id/.default"],
+      "RequestAppToken": true
+    }
+  }
+}
+```
+
+### ASP.NET Core Setup
+
+```csharp
+using Microsoft.Identity.Web;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add authentication
+builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAd"))
+    .EnableTokenAcquisitionToCallDownstreamApi()
+    .AddInMemoryTokenCaches();
+
+// Register downstream APIs
+builder.Services.AddDownstreamApi("MyApi",
```

Comment:
Instead of 2 AddDownstreamApi, we should have one:

```csharp
builder.Services.AddDownstreamApis(builder.Configuration.GetSection("DownstreamApis"));
```